src/PKSim.Infrastructure/ORM/Queries/ModelObserverQuery.cs

```diff
@@ -261,6 +282,25 @@ private IFormula getFractionOfDoseFormula(IObserverBuildingBlock observerBuildin
          return getFractionFormula(observerBuildingBlock, formulaName, compound.Name);
       }
 
+      private IFormula getTotalFractionOfDoseFormula(IObserverBuildingBlock observerBuildingBlock, Compound compound, string fractionOfDoseObserverName)
+      {
+         var formulaName = $"TotalFractionOfDose_{compound.Name}";
+
+         if (observerBuildingBlock.FormulaCache.Contains(formulaName))
+            return observerBuildingBlock.FormulaCache[formulaName];
+
+         var sumFormula = _objectBaseFactory.Create<SumFormula>()
+            .WithName(formulaName)
+            .WithDimension(_dimensionRepository.Fraction);
+
+         //This will collect all observer named 'fractionOfDoseObserverName' for the molecule where the observer is defined.
+         sumFormula.Criteria = Create.Criteria(x => x.With(fractionOfDoseObserverName).And.InContainer(ObjectPathKeywords.MOLECULE));
```

Comment:
That's my updated criteria. It's going to collect ALL observers named "fractopnOfDoseObserverName" in a container MOLECULE!! BOOM